### PR TITLE
TexturePacker: some cleanup and win32 buildsystem fixes

### PR DESCRIPTION
**Pull request by Montellese**
This replaces #7839 (thanks @uNiversaI) and adds two additional (non-critical) fixes to the TexturePacker. The first commit makes sure that running TexturePacker multiple times with the same input always produces the same output. The second commit fixes the debug build configuration in the VS solution.

### CODE REVIEW

**Comment by uNiversaI** [comment]
yw.

Question: Isnt Texturepacker able to compile independently from Kodi?
I cant seem to be able to compile Texturepacker without compiling squish (only) from the Kodi project first.

**Comment by Montellese** [comment]
It can't compile independently from Kodi because it shares some source files as well. But those don't have to be pre-built. I never checked the other dependencies but I guess they have to be there already as well.

**Comment by uNiversaI** [comment]
Well since I never compiled Kodi here, So with some certainty can claim only squish seems to be required to be built previously to compiling texturepacker. Not a big deal but good to know.

Thx.

**Comment by mkortstiege** [comment]
jenkins build this please

File: tools/depends/native/TexturePacker/src/XBTFWriter.cpp [hunk 1]
```diff
@@ -116,12 +117,15 @@ bool CXBTFWriter::UpdateHeader(const std::vector<unsigned int>& dupes)
   {
     CXBTFFile& file = files[i];
 
-    // Convert path to lower case
+    // Convert path to lower case and store it into a fixed size array
     std::string path = file.GetPath();
+    char pathMem[CXBTFFile::MaximumPathLength];
+    memset(pathMem, 0, sizeof(pathMem));
+
     for (std::string::iterator ch = path.begin(); ch != path.end(); ++ch)
```
Comment (Paxxi) [inline tools/depends/native/TexturePacker/src/XBTFWriter.cpp]: Maybe replace with
    std::transform(s.begin(), s.end(), s.begin(), ::tolower);

Comment (Montellese) [inline tools/depends/native/TexturePacker/src/XBTFWriter.cpp]: We need a memory which has a size of 256 bytes and contains the path. I'm not sure why it was done this way but that's how the old texturepacker worked so we need to keep that for backwards compatibility.

Comment (Paxxi) [inline tools/depends/native/TexturePacker/src/XBTFWriter.cpp]: Maybe stick a comment there so we know the next time someone touches this

Comment (Montellese) [inline tools/depends/native/TexturePacker/src/XBTFWriter.cpp]: Done (a few lines further up).

**Comment by Montellese** [comment]
I've expanded the comment on how to write the path into the XBT file as requested by @Paxxi. jenkins build this please

**Comment by Montellese** [comment]
How do we want to handle this with the prebuilt win32 version? We can't update the already uploaded archive with the 1.0.3 version. We can either bump the version to 1.0.4 in the project file and upload a new package with that version or leave the version at 1.0.3 (as fixed by this PR) and upload a new package with 1.0.3-1. @MartijnKaijser preferences?

**Comment by MartijnKaijser** [comment]
just bump it as that makes most sense.

**Comment by Montellese** [comment]
Bumped to 1.0.4 but the package is missing (can't upload right now). Will do later and then merge.

**Comment by Montellese** [comment]
jenkins build and merge

**Pull request merged**